[Clif] ip

Clifong opened the pull request:
# This is my pull requests
## This is my pull request
### This is my pull request

- This is my pull request
- This is my pull request
- This is my pull request

1. This is my pull request
2. This is my pull request
3. This is my pull request

This is my [pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests)

This is my `PullRequest`

```java
function PullRequest() {
  console.log("This is my pull request");
}
```

- [X] This is my pull **request**
- [ ] This is my _pull_ request
- [ ] This is my pull request

This is my pull request 😡

> This is my pull request

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Clifong committed:
Rename main class

Main.java becomes Acheron.java

Clifong committed:
Add greeting

Changed greeting

Clifong committed:
Add support for echo

Echo function now available

Clifong committed:
Add support for list

Clifong committed:
added mark and unmark

added ability to mark and unmark tasks

Clifong committed:
added todos, deadlines and events

Added todos, deadlines and events

Clifong committed:
used automated testing

Used automated testing

Clifong committed:
added handle error

Added ability to handle error

Clifong committed:
added delete

Added delete

Clifong committed:
added saving

Added saving feature

Clifong committed:
add date time resolver

Added function to format and recognise date

Clifong committed:
added more OOP

Seperaed logic further into their own class

Clifong committed:
refactored all directories into a single package

Refactored all necessary directories into a single package

Clifong committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Clifong committed:
added some junit tests

Added some junix tests

Clifong committed:
created a jar file

Cretaed a jar file

Clifong committed:
added javadoc comments

Added some javadoc comments to know what the class and function does

Clifong committed:
changed code to reflect coding standard

Change some code to ensure coding standard is uphold

Clifong committed:
added find function

Added ability to find tasks using a keyword

Clifong committed:
Merge branch 'branch-A-CodingStandard'

Clifong committed:
update duke.txt

Updated duke.txt

Clifong committed:
Merge branch 'branch-Level-9'

Clifong committed:
Modify comment

Made the comment for findAllTaskWithKeyword(String keywork) clearer

Clifong committed:
Add comments

Add comments for StorageManagerNonexistentPath() test

Clifong committed:
Add comments for test

Add comments StorageManagerNonsensicalTask() for clarity

Clifong committed:
Add comments for test

Add comments to TaskWriterTodoEmptySpace() for clarity

Clifong committed:
Add comments for test

Add comments to TaskWriterTodoEmpty() for clarity

Clifong committed:
Add comments to test

Add comments to TaskWriterDeadlineWrongFormat() for clarity

Clifong committed:
Add comments to test

Add comments to TaskWriterDeadlineWrongLengthButRightFormat() to make it clearer

This test checks if the task writer receives the wrong number
of characters but has the right format for the date

Makes it easier to understand.

Clifong committed:
Add comments for task

Add comments to TaskWriterEventWrong() to make it clearer

This test checks if the task writer receives the wrong format for the input

Makes it easier to understand.

Clifong committed:
Add comment for test

Add comments to TaskWriterSavedFileTodoEmptySpace() to make it clearer

This test checks if the task writer receives a wrong line of text
in the saved file

Makes it easier to understand.

Clifong committed:
Add comment for test

Add comments to TaskWriterSavedFileTodoEmpty() to make it clearer

This test checks if the task writer receives a wrong line of text
in the saved file

Makes it easier to understand.

Clifong committed:
Add comments for test

Add comments to TaskWriterSavedFileDeadlineWrongFormat() to make it clearer

This test checks if the task writer receives a wrong line of text
in the saved file

Makes it easier to understand.

Clifong committed:
Add comment for test

Add comments to TaskWriterSavedFileDeadlineWrongLengthButRightFormat() to make it clearer

This test checks if the task writer receives a wrong line of text
in the saved file

Makes it easier to understand.

Clifong committed:
Add comments to test

Add comments to TaskWriterSavedFileEventWrong() to make it clearer

This test checks if the task writer receives a wrong line of text
in the saved file

Makes it easier to understand.

Clifong committed:
Update style

Update style of code to be correct.

As per the requirements of the branch, some stuff
modified include trailing white spaces, naming style
and so on.

Check style before committing

Clifong committed:
Add GUI

Add the GUI for duke.

The GUI allows user to input a text and receive
a response accordingly.

This is done using javafx. Be careful next time
when trying to update the UI.

Clifong committed:
Change some function

Add a return statement for some function.

Because of how the code is written, an error
is thrown if none of the if statements are
satisfied.

This cause an error to be thrown even though
the command is valid. The return statement helps
to prevent the error part to be thrown.

Clifong committed:
Add some assertions

Add some assertions.

The assertions check if there are any weird
anomalies. That is to say there must be
things we expect to be in a certain format,
or with a certain value must hold true.
Otherwise something is wrong with the
program for unknown reasons or
something is wrong with the user's device.

Clifong committed:
Change code quality

Change the quality of some lines of code

Some changes include improving the
method name so they are clearer. Some
 changes include changing the java
docs comment for better quality. Other
changes include reformatting some text
to improve readibility

Clifong committed:
Change code readebility

More improvement to the code.

Some of the changes include
modifying the name of some files
so they are clearer. Some of the
changes include adding missing
text. Some of the changes also
include rectifying some spelling
mistakes

Clifong committed:
Merge pull request #1 from Clifong/branch-A-Assertions

Add some assertions

Clifong committed:
Merge branch 'master' into branch-A-CodeQuality

Clifong committed:
Merge pull request #2 from Clifong/branch-A-CodeQuality

Branch a code quality

Clifong committed:
Clean up code

Clean up the code a bit.

Ensure that useless or incorrect
imports or variables are removed.
This is to ensure that in future
development, there will not be any
confusion on what these extra stuff
are.

Clifong committed:
Add command info classes

Add command info classes

These classes ae used to show what
commands the user can use, how
the command should be and what
the commands do.

Also, users can now key in help to
see these commands so at least they
know what the chatbot does.

DiligentPenguinn commented on src/main/java/Acheron/Tasks/Deadlines.java:
```diff
@@ -0,0 +1,45 @@
+     * @return The content of the task
+     */
+    @Override
+    public String saveTask(boolean last) {
```
Perhaps `isLast` would be a better name for boolean?

Clifong replied:
Okay

DiligentPenguinn commented on src/main/java/Acheron/Tasks/Events.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class Events extends Tasks {
+
+    private LocalDate from;
+    private LocalDate to;
```
Do you think nouns would make more meaningful names here (e.g. startTime, endTime)?

Clifong replied:
Okay

DiligentPenguinn commented on src/main/java/Acheron/Tasks/Deadlines.java:
```diff
@@ -0,0 +1,45 @@
+/**
+ * Represents a deadlines task
+ */
+public class Deadlines extends Tasks {
```
Would singular nouns (Task, Deadline) be more intuitive for these classes' names?

Clifong replied:
Okay

DiligentPenguinn commented on src/main/java/Acheron/UI/UI.java:
```diff
@@ -0,0 +1,29 @@
+/**
+ * Represents the UI display
+ */
+public class UI {
```
Maybe `Ui` would adhere to the coding standard more

Clifong replied:
Okay

DiligentPenguinn reviewed (commented):
Overall, I found your code follow the coding standard well. Also I like the way you structure everything, very neat!

Clifong committed:
Change some details

Change some naming of the files and variables
to be more in line with the coding standard.

Most of the change was due to the pull request
review where reviewers have helped to state
some improvements to the naming conventions
used.

Also, I have changed the structure of the files
and removed some unnecessary variables or
functions.

Clifong committed:
Change some details

Change some naming of the files and variables
to be more in line with the coding standard.

Most of the change was due to the pull request
review where reviewers have helped to state
some improvements to the naming conventions
used.

Also, I have changed the structure of the files
and removed some unnecessary variables or
functions.

Clifong committed:
Change some details

Change some naming of the files and variables
to be more in line with the coding standard.

Most of the change was due to the pull request
review where reviewers have helped to state
some improvements to the naming conventions
used.

Also, I have changed the structure of the files
and removed some unnecessary variables or
functions.

Clifong committed:
Fix some errors

Fix some bugs.

In the deadline and event command info,
the text used erroneously used \ instead of
/. For example, itshould be /by XXXX instead
of \by XXXX. I have changed this part so users
will be shown the correct input format.

Clifong committed:
Change user guide

The README is now updated with all information
that is critical to allow the user to understand all
the major features the chatbot offers. Anyone can
freely read it now

Add user guide

Add user guide

The README is now updated with all information
that is critical to allow the user to understand all
the major features the chatbot offers. Anyone can
freely read it now

Clifong committed:
Fix some bugs

Fix some bugs.

In the deletion of task, the task number
may not be valid. There may not be a check
used to ensure proper input. I have not added
a check to tell the user to reinput the correct
command.

Also, the remove command is also added.
This is to ensure people know about the
existence of a remove command.

Clifong committed:
Change code quality

Update the code quality by changing some
of the methods name and comments. Also
changed some of the method name to ensure
they run correctly.

Refactored some code as well to ensure that
SLAP is more adhered to.
